Generator/ProductGenerator.php

```diff
@@ -199,6 +220,10 @@ public function generate(array $config, $outputDir, ProgressHelper $progress, ar
                 }
             }
 
+            if (null !== $variantGroupDataProvider) {
+                $product = array_merge($product, $variantGroupDataProvider->getData());
+            }
```

Comment:
How does it work? let say i generate 1M products and 10k variant groups with a 10 product per variant group config.
All the first products are attached to a variant group then once they're all fullfilled last products have no variant group?
Maybe add a early check to raise an exception when "number of variant group x number of products per variant group > number of products" ?